### PR TITLE
Labs-API Integration

### PR DESCRIPTION
**Pull request by agordon1123**
# Description

- Updated analytics to fire upon location change
- Populated labs-api database with list of labs 17 products
- Connected DashboardHome to the redux store and called getProducts on mount
- Created handleSearch function to dynamically render list of products based on search params

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Change Status

- [X] Complete, but not tested (~may need~ **definitely needs** new tests)

# How Has This Been Tested?

- [X] Smashing on my keyboard in the search bar
image

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My code has been reviewed by at least one peer
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by agordon1123** [comment]
Hey all,

I started this branch yesterday and copied Dan's code for the inputs from his PR. It looks like it's been updated since then so feel free to decline any of my code dealing with the auth components when handling the merge conflicts.

**Review by armandoroman1016 — COMMENTED** [review]
Everything looks good here, awesome job on the search function 👍

**Review by armandoroman1016 — DISMISSED** [review]

**Review by danbergelt — COMMENTED** [review]
@agordon1123 There are merge conflicts, would you be able tor resolve so we can all review?

File: mission-control-client/src/App.js [hunk 1]
```diff
@@ -8,13 +8,13 @@ import Login from './components/auth/Login';
...
-import Login from './components/auth/Login'
 
 function App() {
-  
+  const location = useLocation();
+
   useEffect(() => {
     embedAnalytics();
-  }, []);
+  }, [location]);
```
Comment (danbergelt) [inline mission-control-client/src/App.js:17]: Reminder that we need to look into this more and make sure that we're taking advantage of full functionality

File: mission-control-client/src/components/dashboard/admin-dashboard/ProductList.js [hunk 2]
```diff
@@ -6,22 +6,28 @@ import { Input } from 'semantic-ui-react'
...
-      items = items.filter(item => {
-        return item.name.toLowerCase().search(e.target.value.toLowerCase()) !== -1;
+    const products = props.products;
+    const re = /^[a-z0-9]+$/i;
+
+    if (e.target.value !== "" && re.test(e.target.value)) {
+      setFiltered({
+        products: products.filter(item => {
+          return item.productName.toLowerCase().search(e.target.value.toLowerCase()) !== -1;
+        })
```
Comment (danbergelt) [inline mission-control-client/src/components/dashboard/admin-dashboard/ProductList.js:25]: Are parts/all of this maybe worth extracting as a util? Would make testing a lot easier.

**Review by danbergelt — APPROVED** [review]

**Review by dionys93 — APPROVED** [review]

**Review by armandoroman1016 — APPROVED** [review]

**Pull request merged**